### DISCUSSION TITLE
Need some dummy data on localhost for leaderboard.

### DISCUSSION THREAD

**Discussion by HyperDanisH** [monkeytypegame/monkeytype#4474] [Questions] [answered]
Hi everyone,

I am working on a feature #4424.

I need some data to fill up leaderboard to play with it. I have tried configuring the backend server's config but nothing seems to work.

Would appericiate some help.

Best.

**Comment by Miodec** [comment, marked as answer]
You can just create a few accounts

**Reply by HyperDanisH** [reply]
Hi, I have just tried what you said, everything works fine till submition of results but after that nothing shows up in leader board.

**Reply by Miodec** [reply]
You need to configure daily leaderboards:

<img width="137" alt="image" src="https://github.com/monkeytypegame/monkeytype/assets/13181393/cf63fedf-259b-4fd7-9f83-5224a38ea8e1">


<img width="751" alt="image" src="https://github.com/monkeytypegame/monkeytype/assets/13181393/769f1d50-ed75-4973-a25a-7c0a19e5ab77">

**Reply by HyperDanisH** [reply]
Hi sir,

I have tried what you have said in this post, but there are some errors popping up, which I think if I ask you would speed up the process.

I have set the configration as you said.

![Screenshot 2023-07-25 102310](https://github.com/monkeytypegame/monkeytype/assets/93310647/8dbc9fc2-8091-4146-bd57-0496c8256eb2)
Note: I have even tried settion up secondary value to (15|60)

It is saved properly in mongodb (the configration part). But now when I save results in 15 secs mode or 60 secs mode there is an error that occurs, this is with redis scripts.

Frontend:
![Screenshot 2023-07-25 102127](https://github.com/monkeytypegame/monkeytype/assets/93310647/6316bec3-9a94-4e82-a892-f26bbe66fcd0)


Backend:
![Screenshot 2023-07-25 102222](https://github.com/monkeytypegame/monkeytype/assets/93310647/537d7253-65eb-4842-be55-a1507b305182)

**Reply by Miodec** [reply]
@Bruception , my Redis wizzard - do you know whats going on here?

**Reply by Bruception** [reply]
Sounds like some value being passed to the daily lb script is not the expected type. I would log the values being passed to debug further.

**Reply by Bruception** [reply]
https://github.com/monkeytypegame/monkeytype/blob/90761fbc329657d4f9ea07f2671da22212350a65/backend/src/utils/daily-leaderboards.ts#L83

**Reply by HyperDanisH** [reply]
Hi @Bruception @Miodec, Did anyone of you got into such error? Maybe it could be error with the operating system (Windows in this case) or maybe I try installing lua. I have tried console logging the values and everything seems fine except this:

![Screenshot 2023-07-26 143138](https://github.com/monkeytypegame/monkeytype/assets/93310647/96b360e2-5ea2-48dc-85ec-49ac6157a215)

Here some details about discord server are undefined, could this be the reason?

**Reply by Bruception** [reply]
Could you log all of the arguments to the addResult function?

**Reply by HyperDanisH** [reply]
Hi,

I can actually log all the arguments, here's a screenshot:

![Screenshot 2023-07-27 131859](https://github.com/monkeytypegame/monkeytype/assets/93310647/5c9b49a1-c145-429d-a132-807a63ff0b25)


![Screenshot 2023-07-27 131819](https://github.com/monkeytypegame/monkeytype/assets/93310647/61f96a1e-53bf-48e9-9b43-54532538b553)

There is definely some problem in `const rank = await connection.addResult(`. I can try two things now (Which I am about to), first is to install lua language locally, second is to switch os, I'll use ubuntu this time instead of Windows.

In the mean time, I would really love to have your views on this one. I apologize for being such a newbie but It's first time I am using redis and I will soon figure things out.

Best,

**Reply by Miodec** [reply]
Weird, everything looks fine.

**Reply by Miodec** [reply]
I logged mine, the only difference it seems is that my wpm and raw are floats 
![image](https://github.com/monkeytypegame/monkeytype/assets/13181393/187b23a5-bdfc-4dec-9791-170f5672845c)

**Reply by Miodec** [reply]
Can you try to add another result to the leaderboard and see if you get floats?

**Reply by Miodec** [reply]
Nevermind, i sent it rounded data and it didnt error
![image](https://github.com/monkeytypegame/monkeytype/assets/13181393/5150456a-02a7-424a-b947-15e1f2a2718b)

**Reply by HyperDanisH** [reply]
I tried everyting and there is nothing that works. Nor switching os did work.

**Reply by Bruception** [reply]
Are you using docker compose to spin up the database and redis server?

**Reply by HyperDanisH** [reply]
No, I am not. I am compiling it locally but should I try so?

**Reply by Miodec** [reply]
Yeah try that

**Reply by HyperDanisH** [reply]
> Yeah try that

Yes, I did that and fortunately spinning docker containers for both mongodb and redis does the job. 

Thanks for the help through the way for you both. I will really soon issue the pr implementing this reature.

Best,
